src/Configuration.php

```diff
@@ -82,12 +95,29 @@ public static function isBooted(): bool
 
     public static function boot(\Closure|self $configuration): void
     {
-        self::$instance = $configuration;
+        self::$instance = \is_callable($configuration) ? ($configuration)() : $configuration;
+        self::$instance->bootedForDataProvider = false;
+    }
+
+    public static function bootForDataProvider(\Closure|self $configuration): void
+    {
+        self::$instance = \is_callable($configuration) ? ($configuration)() : $configuration;
+        self::$instance->bootedForDataProvider = true;
     }
 
     public static function shutdown(): void
     {
         StoryRegistry::reset();
         self::$instance = null;
     }
+
+    public function enableInMemory(): void
+    {
+        $this->inMemory = true;
+    }
+
+    public function isInMemoryEnabled(): bool
+    {
+        return $this->inMemory;
+    }
```

Comment:
At first, I did something like:
```php
Configuration::instance()->persistence()->enableInMemory();
```

and the `$inMemory` boolean was stored in `PersistenceManager`. But then, all `InMemoryTest` did not worked when no persistence is activated, which is a non-sense :shrug: 

Then I decided to store this in the `Configuration` class